src/librustc_middle/mir/mono.rs

```diff
@@ -229,6 +204,19 @@ impl<'a, 'tcx> HashStable<StableHashingContext<'a>> for MonoItem<'tcx> {
     }
 }
 
+impl<'tcx> fmt::Display for MonoItem<'tcx> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        ty::tls::with(|tcx| match tcx.lift(self).expect("could not lift for printing") {
+            MonoItem::Fn(instance) => {
+                let name = tcx.def_path_str_with_substs(instance.def_id(), instance.substs);
+                write!(f, "fn {}", &name)
```

Comment:
My suggestion still stands, doesn't `Instance` already implement `fmt::Display`?

Reply:
It does but that adds a prefix to the output, e.g.

```
These items should have been contained but were not:

MONO_ITEM fn std::intrinsics::drop_in_place::<LocalStruct> @@ extern_drop_glue-fallback.cgu[External]
MONO_ITEM fn std::intrinsics::drop_in_place::<cgu_extern_drop_glue::Struct> @@ extern_drop_glue-fallback.cgu[External]
MONO_ITEM fn std::intrinsics::drop_in_place::<mod1::LocalStruct> @@ extern_drop_glue-fallback.cgu[External]



These items were contained but should not have been:

MONO_ITEM fn std::intrinsics::drop_in_place::<LocalStruct> - shim(Some(LocalStruct)) @@ extern_drop_glue.3a1fbbbh-fallback.cgu[External]
MONO_ITEM fn std::intrinsics::drop_in_place::<cgu_extern_drop_glue::Struct> - shim(Some(cgu_extern_drop_glue::Struct)) @@ extern_drop_glue.3a1fbbbh-fallback.cgu[External]
MONO_ITEM fn std::intrinsics::drop_in_place::<mod1::LocalStruct> - shim(Some(mod1::LocalStruct)) @@ extern_drop_glue.3a1fbbbh-fallback.cgu[External]
```

that I'm not sure we'd want?

Reply:
Do you mean suffix? Because I'd definitely want that, right now that's information present in the `Instance`, being hidden.
This is more important for e.g. reification shims, which can coexist alongside a non-shim original definition.

Reply:
Fixed!